pkg/adapter/mtping/runner.go

```diff
@@ -122,9 +125,21 @@ func (a *cronJobsRunner) Stop() {
 
 func (a *cronJobsRunner) cronTick(ctx context.Context, event cloudevents.Event) func() {
 	return func() {
+
+		event := event.Clone()
+		event.SetID(uuid.New().String())
```

Comment:
I actually wonder why this is needed. The CE SDK adds this automatically.

Reply:
I've not seen it there.
Can you do a getId with out this and log .. see if it's there.

Reply:
I've also seen where other adapters are setting it.

Reply:
@lionelvillard  You are right if  you don't explicitly set the ID the sdk client will provide one. However it seems you can set it here explicitly and I have verified the sink gets this ID in the the event.  The reason for doing it here it allows us before the event starts on in it's journey to the sink we can log it in debug and report it in an error. I think this is IMO invaluable in trying diagnosing  issues.

Reply:
make sense! Thanks!